### DISCUSSION TITLE
How to use client function in pinia store

### DISCUSSION THREAD

**Discussion by JoshPJackson** [manchenkoff/nuxt-auth-sanctum#219] [Q&A] [answered]
I have a pinia store in which I would like to retrieve some models from my api. Within the store actions I have a method to get all pages:

```
import { defineStore } from 'pinia'
const client = useSanctumClient()

export const usePageStore = defineStore({
  id: 'myPageStore',
  state: () => ({ }),
  actions: {
    get(id: string) {
      return client('/api/pages/' + id)
    },

    getAll() {
      return client('/api/pages')
    }
  }
})
```

However when loading the page in the browser I see this in the console:
`client is not a function`.

Is there a nice way of solving this issue or do I just need to move the api calls into the components?

**Comment by manchenkoff** [comment, marked as answer]
Hey @JoshPJackson! Thanks for the question. Indeed you can use `useSanctumClient` in your Pinia store, but it looks like that the scope of the call is incorrect. Since `nuxt-auth-sanctum` module requires Nuxt context to be present, you should move variable instantiation inside of the store, something like this:

```typescript
import { defineStore } from 'pinia'

export const usePageStore = defineStore('myPageStore', () => {
  const client = useSanctumClient()
  
  async function get(id: string) {
    return await client('/api/pages/' + id)
  }

  async function getAll() {
    return await client('/api/pages')
  }

  return {get, getAll}
})
```
